flask_praetorian/base.py

```diff
@@ -484,3 +550,111 @@ def pack_header_for_user(
             **custom_claims
         )
         return {self.header_name: self.header_type + ' ' + token}
+
+    def send_registration_email(self, user=None, template=None,
+                                subject=None,
+                                **kwargs):
+        """
+        Sends a registration email to a new user, containing a time expiring
+            token usable for validation.
+        :param: user:                     The user object to tie claim to
+                                          (username, id, email, etc)
+        :param: subject:                  The registration email subject
+                                          override.
+        :param: override_access_lifepsan: Anything used in `encode_jwt_token`,
+                                          but all we really care about is
+                                          `override_access_token`.  This
+                                          overrides the instance's defined
+                                          lifespan, which is used as the
+                                          default.  The token will no longer
+                                          be valid after this time and
+                                          be rejected.
+        :param: template:                 Template file location for email.
+                                          If not provided, a stock entry is
+                                          used.
+        """
+
+        class Notification:
+            def __init__(self):
+                self.errors = None
+                self.message = None
+                self.email = user.email
+                self.token = None
+                self.confirmation_uri = None
+
+        notification = Notification()
+
+        with PraetorianError.handle_errors('fail sending confirmation email'):
+            notification.token = self.encode_jwt_token(user, **kwargs)
+            _confirmation_uri = url_for(self.confirmation_endpoint,
+                                        _external=True)
+            notification.confirmation_uri = '/'.join([_confirmation_uri,
+                                                      notification.token])
+
+            with open(self.email_template) as _template:
+                tmpl = Template(_template.read())
+            notification.message = tmpl.render(notification.__dict__).strip()
+            subject = subject if subject else self.confirmation_subject
+
+            msg = Message(body=notification.message,
```

Comment:
```suggestion
            msg = Message(
                body=notification.message,                          
                sender=self.confirmation_sender,
                subject=subject,
                recipients=[notification.email],
            )
```

Reply:
submitted

Reply:
i jacked this up working on two branchs.  i'm closing both, resubmitting as a single pull